coderd/coderdtest/testjar/cookiejar.go

```diff
@@ -0,0 +1,33 @@
+package testjar
+
+import (
+	"net/http"
+	"net/url"
+	"sync"
+)
+
+func New() *Jar {
+	return &Jar{}
+}
+
+// Jar exists because 'cookiejar.New()' strips many of the http.Cookie fields
+// that are needed to assert. Such as 'Secure' and 'SameSite'.
+type Jar struct {
+	m      sync.Mutex
+	perURL map[string][]*http.Cookie
```

Comment:
you could use `syncmap.Map` since this is the only thing being locked by the mutex
```suggestion
	perURL syncmap.Map[string, []*http.Cookie]
```

Reply:
Unfortunately I cannot.
I need to append to the existing value. Using a syncmap would require me to `Load()`, then `append()`. Which has a window between the two calls another thread could change the underlying slice.

If syncmap had some api to allow me to do all that in an atomic function, that would be cool.